app/gameengine.py

```diff
@@ -3889,14 +4017,18 @@ def archive_hand_continuation():
                     })
             case EffectType.EffectType_RevealTopDeck:
                 if len(effect_player.deck) > 0:
-                    top_card = effect_player.deck[0]
+                    amount = effect.get("amount", 1)
+                    top_cards = effect_player.deck[:amount]
+                    effect_player.last_revealed_cards = top_cards
                     reveal_event = {
                         "event_type": EventType.EventType_RevealCards,
                         "effect_player_id": effect_player_id,
-                        "card_ids": [top_card["game_card_id"]],
+                        "card_ids": ids_from_cards(top_cards),
                         "source": "topdeck"
                     }
                     self.broadcast_event(reveal_event)
+                    after_reveal_effects = effect_player.get_effects_at_timing("after_reveal", None)
+                    self.add_effects_to_rear(after_reveal_effects)
```

Comment:
i think you want to add to front?
if there were a bunch of queued up AND effects, it could get weird if the after reveal occurs after all those.